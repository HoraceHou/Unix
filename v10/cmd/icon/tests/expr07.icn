record array(a,b,c,d,e,f,g)

procedure dummy(u,v,x,y,z)
   suspend u | v
   return x
end

procedure main()
   write("t := table() ----> ",image(t := table()) | "none")
   write("every t[1 | 2.0 | &output | &cset | [] | \"hello\" | main | 2 | table()] := ?100 ----> ",image(every t[1 | 2.0 | &output | &cset | [] | "hello" | main | 2 | table()] := ?100) | "none")
   write("a := sort(t,3) ----> ",image(a := sort(t,3)) | "none")
   write("every i := 1 to *a - 1 by 2 do write(image(a[i]),\" \",a[i + 1]) ----> ",image(every i := 1 to *a - 1 by 2 do write(image(a[i])," ",a[i + 1])) | "none")
   write("a := sort(t,4) ----> ",image(a := sort(t,4)) | "none")
   write("every i := 1 to *a - 1 by 2 do write(image(a[i]),\" \",a[i + 1]) ----> ",image(every i := 1 to *a - 1 by 2 do write(image(a[i])," ",a[i + 1])) | "none")
end
