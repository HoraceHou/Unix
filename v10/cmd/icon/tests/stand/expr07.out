t := table() ----> table(0)
every t[1 | 2.0 | &output | &cset | [] | "hello" | main | 2 | table()] := ?100 ----> none
a := sort(t,3) ----> list(18)
1 22
2 74
2.0 42
"hello" 31
&cset 52
&output 32
procedure main 8
list(0) 43
table(0) 6
every i := 1 to *a - 1 by 2 do write(image(a[i])," ",a[i + 1]) ----> none
a := sort(t,4) ----> list(18)
table(0) 6
procedure main 8
1 22
"hello" 31
&output 32
2.0 42
list(0) 43
&cset 52
2 74
every i := 1 to *a - 1 by 2 do write(image(a[i])," ",a[i + 1]) ----> none
